test/masq.test.js

```diff
@@ -555,3 +578,50 @@ test('should set a watcher', async (done) => {
   masq.watch('/hello', onChange)
   await masq.put(key, value)
 })
+
+test('should be able to get a notif on change in masq-app with a watcher on masq-lib', async (done) => {
+  expect.assertions(2)
+  let resolvePr1
+  const pr1 = new Promise((resolve) => { resolvePr1 = resolve })
+  let resolvePr2
+  const pr2 = new Promise((resolve) => { resolvePr2 = resolve })
+
+  const onChangeMasqLib = () => {
+    expect(true).toBe(true)
+    resolvePr1()
+  }
+  const onChangeMockMasqApp = () => {
+    expect(true).toBe(true)
+    resolvePr2()
+  }
+
+  await logInWithMasqAppMock()
+  const key = '/hello'
+  const value = { data: 'world' }
+  mockMasqApp.watch(masq.userId, '/hello', onChangeMockMasqApp)
```

Comment:
I don't think this is necessary? You just need to focus on the behavior of masq-lib

Reply:
It is the only way to make sure that the two-way replication is done correctly in tests.

Reply:
Yes we could remove the promise on mocked masq app

Reply:
done !